File: app/src/main/java/com/example/android/simpledice/ui/HistoryResultsAdapter.java
package com.example.android.simpledice.ui;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.android.simpledice.utils.DiceResults;
import com.example.android.simpledice.R;

import java.util.List;

public class HistoryResultsAdapter extends RecyclerView.Adapter<HistoryResultsAdapter.HistoryResultsViewHolder> {

    private List<DiceResults> mHistoryResults;

    private final HistoryResultsClickHandler mHistoryResultsClickHandler;

    HistoryResultsAdapter(HistoryResultsClickHandler historyResultClickHandler) {
        mHistoryResultsClickHandler = historyResultClickHandler;
    }


    //Inner ViewHolder Class
    public class HistoryResultsViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        TextView mNameTextView;
        TextView mDateTextView;
        TextView mTotalTextView;

        //Constructor
        public HistoryResultsViewHolder(@NonNull View itemView) {
            super(itemView);

            mNameTextView = itemView.findViewById(R.id.history_name_tv);
            mDateTextView = itemView.findViewById(R.id.history_date_tv);
            mTotalTextView = itemView.findViewById(R.id.history_total_tv);

            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            int adapterPosition = getAdapterPosition();
            DiceResults diceResults = mHistoryResults.get(adapterPosition);
            mHistoryResultsClickHandler.onItemClick(diceResults);
        }
    }

    @NonNull
    @Override
    public HistoryResultsViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        Context context = viewGroup.getContext();
        int layoutIDForListItem = R.layout.history_recycler_item;
        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(layoutIDForListItem, viewGroup, false);

        return new HistoryResultsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull HistoryResultsViewHolder viewHolder, int position) {
        DiceResults currentDiceResults = mHistoryResults.get(position);

        String name = currentDiceResults.getName();
        String date = currentDiceResults.getFormattedDateCreated();
        String total = String.valueOf(currentDiceResults.getTotal());

        viewHolder.mNameTextView.setText(name);
        viewHolder.mDateTextView.setText(date);
        viewHolder.mTotalTextView.setText(total);
    }

    @Override
    public int getItemCount() {
        if (mHistoryResults == null) return 0;
        return mHistoryResults.size();
    }

    /**
     * Helper method to set DiceResults List to existing HistoryResultAdapter
     *
     * @param historyResults the new set of DiceResults to be displayed
     */

    public void setHistoryResults(List<DiceResults> historyResults) {
        mHistoryResults = historyResults;
        notifyDataSetChanged();
    }

    //Interface to handle clicks, defined in MainActivity
    public interface HistoryResultsClickHandler {
        void onItemClick(DiceResults historyResults);
    }
}

File: SimpleDice/app/src/main/java/com/gmail/maxfixsoftware/simpledice/ui/HistoryResultsAdapter.kt
package com.gmail.maxfixsoftware.simpledice.ui

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.gmail.maxfixsoftware.simpledice.R
import com.gmail.maxfixsoftware.simpledice.utils.DiceResults

class HistoryResultsAdapter internal constructor(private val mHistoryResultsClickHandler: HistoryResultsClickHandler) :
    androidx.recyclerview.widget.RecyclerView.Adapter<HistoryResultsAdapter.HistoryResultsViewHolder>() {

    private var mHistoryResults: List<DiceResults>? = null


    //Inner ViewHolder Class
    inner class HistoryResultsViewHolder
        (
        itemView: View,
        var mNameTextView: TextView? = null,
        var mDateTextView: TextView? = null,
        var mTotalTextView: TextView? = null
    ) : androidx.recyclerview.widget.RecyclerView.ViewHolder(itemView), View.OnClickListener {

        init {

            mNameTextView = itemView.findViewById(R.id.history_name_tv)
            mDateTextView = itemView.findViewById(R.id.history_date_tv)
            mTotalTextView = itemView.findViewById(R.id.history_total_tv)

            itemView.setOnClickListener(this)
        }

        override fun onClick(v: View) {
            val adapterPosition = adapterPosition
            val diceResults = mHistoryResults!![adapterPosition]
            mHistoryResultsClickHandler.onItemClick(diceResults)
        }
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): HistoryResultsViewHolder {
        val context = viewGroup.context
        val layoutIDForListItem = R.layout.history_recycler_item
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(layoutIDForListItem, viewGroup, false)

        return HistoryResultsViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: HistoryResultsViewHolder, position: Int) {
        val currentDiceResults = mHistoryResults!![position]

        val name = currentDiceResults.name
        val date = currentDiceResults.formattedDateCreated
        val total = currentDiceResults.total.toString()

        viewHolder.mNameTextView!!.text = name
        viewHolder.mDateTextView!!.text = date
        viewHolder.mTotalTextView!!.text = total
    }

    override fun getItemCount(): Int {
        return if (mHistoryResults == null) 0 else mHistoryResults!!.size
    }

    /**
     * Helper method to set DiceResults List to existing HistoryResultAdapter
     *
     * @param historyResults the new set of DiceResults to be displayed
     */

    fun setHistoryResults(historyResults: List<DiceResults>) {
        mHistoryResults = historyResults
        notifyDataSetChanged()
    }

    //Interface to handle clicks, defined in MainActivity
    interface HistoryResultsClickHandler {
        fun onItemClick(historyResults: DiceResults)
    }
}

